crates/apollo-compiler/src/validation/operation.rs

```diff
@@ -46,6 +46,24 @@ pub(crate) fn validate_subscription(
                 },
             );
         }
+
+        // first rule validates that we only have single selection
+        let has_conditional_selection =
+            operation.selection_set.selections.iter().find(|selection| {
```

Comment:
Sorry for the back-and-forth, but now this lacks spec algorithm’s recursion and fails to catch something like `subscription { ... { field @skip } }`. I’ll move the check into `expand_selections`